sdk/ai/ai-projects/README.md

```diff
@@ -1,57 +1,1045 @@
-provide an instance of the desired credential type obtained from the
-[@azure/identity](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) library.
+<!-- TODO: Update this (?) -->
 
-To authenticate with AAD, you must first `npm` install [`@azure/identity`](https://www.npmjs.com/package/@azure/identity) 
+To connect to Aspire Dashboard or another OpenTelemetry compatible backend, install OTLP exporter:
 
-After setup, you can choose which type of [credential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) from `@azure/identity` to use.
-As an example, [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential)
-can be used to authenticate the client.
+```bash
+pip install opentelemetry-exporter-otlp
+```
 
-Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
-AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET
+<!-- TODO: review/revise tracing example -->
+
+#### Tracing example
+
+Here is a code sample to be included above `createAgent`:
+
+<!-- SNIPPET:sample_agents_basics_with_azure_monitor_tracing.enable_tracing -->
+
+```javascript
+import { trace } from "@opentelemetry/api";
+import { AzureMonitorTraceExporter } from "@azure/monitor-opentelemetry-exporter"
+import {
+    ConsoleSpanExporter,
+    NodeTracerProvider,
+    SimpleSpanProcessor,
+} from "@opentelemetry/sdk-trace-node";
+
+const provider = new NodeTracerProvider();
+provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
+provider.register();
+
+const tracer = trace.getTracer("Agents Sample", "1.0.0");
+
+const client = AIProjectsClient.fromConnectionString(
+  connectionString || "", new DefaultAzureCredential()
+);
+
+if (!appInsightsConnectionString) {
+  appInsightsConnectionString = await client.telemetry.getConnectionString();
+}
+
+if (appInsightsConnectionString) {
+  const exporter = new AzureMonitorTraceExporter({
+    connectionString: appInsightsConnectionString
+  });
+  provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+}
+
+await tracer.startActiveSpan("main", async (span) => {
+    client.telemetry.updateSettings({enableContentRecording: true})
+...
+```
+
+<!-- END SNIPPET -->
 
 ## Troubleshooting
 
+### Exceptions
+
+Client methods that make service calls raise an [HttpResponseError](https://learn.microsoft.com/python/api/azure-core/azure.core.exceptions.httpresponseerror) exception for a non-success HTTP status code response from the service. The exception's `status_code` will hold the HTTP response status code (with `reason` showing the friendly name). The exception's `error.message` contains a detailed message that may be helpful in diagnosing the issue:
```

Comment:
Refer to https://learn.microsoft.com/en-us/javascript/api/%40azure/core-rest-pipeline/resterror